### PR TITLE
STCOM-395: Add routing for holdings and item creation pages

### PR DESCRIPTION
**Pull request by Yurii-Danylenko**
## Purpose
Add a possibility to go to the holdings and item creation pages by route path

## Approach
- Add `layer=createItem` query param which appears after click on "Add new item" button .
- Bind layer to the item creation page to this query param.
- Move layers components outside of the `ViewInstance` component general markup to be rendered like independent pages

### CODE REVIEW

File: src/ItemsPerHoldingsRecord.js [hunk 1]
```diff
@@ -55,59 +54,121 @@ class ItemsPerHoldingsRecord extends React.Component {
...
-
     const formatMsg = this.props.stripes.intl.formatMessage;
-    const viewHoldingsButton = <Link to={viewHoldingsPath}><Button id="clickable-view-holdings">{formatMsg({ id: 'ui-inventory.viewHoldings' })}</Button></Link>;
-    const newItemButton = <Button id="clickable-new-item" onClick={this.onClickAddNewItem} title={formatMsg({ id: 'ui-inventory.addItem' })} buttonStyle="primary paneHeaderNewButton">{formatMsg({ id: 'ui-inventory.addItem' })}</Button>;
     const labelLocation = holdingsRecord.permanentLocationId ? locationsById[holdingsRecord.permanentLocationId].name : '';
     const labelCallNumber = holdingsRecord.callNumber || '';
 
+    if (query.layer === 'createItem') {
+      return (
+        <Layer
```
Comment (cherewaty) [inline src/ItemsPerHoldingsRecord.js]: I'm wondering - if these will be rendered at separately routed URLs - do we still need to wrap it in a `<Layer>` component?

Comment (Yurii-Danylenko) [reply src/ItemsPerHoldingsRecord.js]: Yes, `Layer` is needed to render the component over the main page and stretched to full width of the browser window.

File: src/ItemsPerHoldingsRecord.js [hunk 2]
```diff
@@ -58,22 +58,30 @@ class ItemsPerHoldingsRecord extends React.Component {
...
+    mutator.query.update({ layer: 'createItem' });
+  };
 
   onClickCloseNewItem = (e) => {
     const { mutator } = this.props;
 
     if (e) e.preventDefault();
 
-    mutator.addItemMode.replace({ mode: false });
+    this.props.mutator.addItemMode.replace({ mode: false });
```
Comment (anastasiaHahanova) [inline src/ItemsPerHoldingsRecord.js]: you have restructured mutator from props, but then use this.props.mutator

File: src/ItemsPerHoldingsRecord.js [hunk 3]
```diff
@@ -145,6 +160,27 @@ class ItemsPerHoldingsRecord extends React.Component {
...
+        onToggle={accordionToggle}
+        label={formatMsg({ id: 'ui-inventory.holdingsHeader' }, { location: labelLocation, callNumber: labelCallNumber })}
+        displayWhenOpen={this.renderButtonsGroup()}
+      >
+        <Row>
+          <Col sm={12}>
+            <this.cItems
+              holdingsRecord={holdingsRecord}
+              instance={instance}
+              parentMutator={this.props.mutator}
```
Comment (anastasiaHahanova) [inline src/ItemsPerHoldingsRecord.js]: you can destructure this
and this.props.stripes above

File: src/ViewInstance.js [hunk 4]
```diff
@@ -286,6 +280,40 @@ class ViewInstance extends React.Component {
       </Button>
     );
 
+    if (query.layer === 'edit') {
+      return (
+        <Layer isOpen label={formatMsg({ id: 'ui-inventory.editInstanceDialog' })}>
```
Comment (anastasiaHahanova) [inline src/ViewInstance.js]: please follow our convention, if there is more then 1 property, place them on a separate line

File: src/ViewInstance.js [hunk 5]
```diff
@@ -286,6 +280,40 @@ class ViewInstance extends React.Component {
...
+            referenceTables={referenceTables}
+            stripes={stripes}
+          />
+        </Layer>
+      );
+    }
+
+    if (query.layer === 'createHoldingsRecord') {
+      return (
+        <Layer isOpen label={formatMsg({ id: 'ui-inventory.addNewHoldingsDialog' })}>
```
Comment (anastasiaHahanova) [inline src/ViewInstance.js]: and here

File: src/ViewInstance.js [hunk 6]
```diff
@@ -665,41 +693,21 @@ class ViewInstance extends React.Component {
           :
           null
         }
+
         { (holdingsrecordid && !itemid) ?
           <this.cViewHoldingsRecord id={id} holdingsrecordid={holdingsrecordid} {...this.props} onCloseViewHoldingsRecord={this.closeViewHoldingsRecord} />
```
Comment (anastasiaHahanova) [inline src/ViewInstance.js]: could you please place these properties on separate lines

File: src/ItemsPerHoldingsRecord.js [hunk 7]
```diff
@@ -115,27 +136,22 @@ class ItemsPerHoldingsRecord extends React.Component {
...
-          </Col>
-        </Row>
-        <br />
-        <Layer key={`itemformlayer_${holdingsRecord.id}`} isOpen={addItemMode ? (addItemMode.mode && this.addItemModeThisLayer) : false} label={formatMsg({ id: 'ui-inventory.addNewHoldingsDialog' })}>
+    if (query.layer === 'createItem') {
+      return (
+        <Layer
+          key={`itemformlayer_${holdingsRecord.id}`}
+          isOpen
+          label={formatMsg({ id: 'ui-inventory.addNewHoldingsDialog' })}
```
Comment (anastasiaHahanova) [inline src/ItemsPerHoldingsRecord.js]: could you please replace formatMsg with `<FormattedMessage>`

File: src/ItemsPerHoldingsRecord.js [hunk 8]
```diff
@@ -145,6 +161,27 @@ class ItemsPerHoldingsRecord extends React.Component {
...
         </Layer>
+      );
+    }
+
+    return (
+      <Accordion
+        open={accordionStates[holdingsRecord.id] === undefined || accordionStates[holdingsRecord.id]}
+        id={holdingsRecord.id}
+        onToggle={accordionToggle}
+        label={formatMsg({ id: 'ui-inventory.holdingsHeader' }, { location: labelLocation, callNumber: labelCallNumber })}
```
Comment (anastasiaHahanova) [inline src/ItemsPerHoldingsRecord.js]: could you please replace formatMsg with `<FormattedMessage>`

**Pull request merged**